app/models/user.rb

```diff
@@ -46,6 +48,15 @@ class User < ActiveRecord::Base
   # We really need a Dispatch Chain here or something.
   # This will also let us return a human error message.
   #
+
+  def self.from_omniauth(auth)
+    where( email: auth.info.email).first_or_create do |user|
+      user.email = auth.info.email
+      user.password = Devise.friendly_token[0,20]
+      user.name = auth.info.name   # assuming the user model has a name
```

Comment:
Please, remove the comment from here. It is an artifact from the code that was taken from internet.